src/Generator/Parsing.fs

```diff
@@ -0,0 +1,199 @@
+        | "GLeglImageOES" -> Some GLeglImageOES
+        | "GLhandleARB" -> Some GLhandleARB
+        | "GLintptrARB" -> Some GLintptrARB
+        | "GLsizeiptrARB" -> Some GLsizeiptrARB
+        | "GLsync" -> Some GLsync
+        | "struct_cl_context" -> Some Struct_cl_context
+        | "struct_cl_event" -> Some Struct_cl_event
+        | "GLDEBUGPROC" -> Some GLDEBUGPROC
+        | "GLDEBUGPROCAMD" -> Some GLDEBUGPROCAMD
+        | "GLDEBUGPROCARB" -> Some GLDEBUGPROCARB
+        | "GLDEBUGPROCKHR" -> Some GLDEBUGPROCKHR
+        | _ ->
+            printfn "input %s is not a valid type" str
+            None
+    tryParse str
+
+
+
+open Types
+open SpecificationOpenGL
+open Util
+
+open System
+open System.IO
+open System.Text.RegularExpressions
+
+let readSpec (path:string) = OpenGL_Specification.Load path
+
+let getEnumsFromSpecification (spec: OpenGL_Specification.Registry) =
+    let valueForName =
+        spec.Enums
+        |> Array.Parallel.collect(fun e ->
+            e.Enums
+            |> Array.Parallel.choose(fun case ->
+                case.Value.String
+                |> Option.map(fun value -> (case.Name, value)))
+        )
+        |> Map.ofArray
+
+    spec.Groups
+    |> Array.Parallel.choose(fun e ->
+        maybe {
+            let group = e.Name
+            let cases =
+                e.Enums
+                |> Array.Parallel.choose(fun case ->
+                    valueForName
+                    |> Map.tryFind case.Name
+                    |> Option.map(fun value ->
+                        { name = case.Name
+                          value =  value })
+                )
+                |> Array.groupBy(fun case -> case.value)
+                |> Array.Parallel.choose(fun (_, cases) ->
+                    cases
+                    |> Array.tryFind(fun case -> (case.name.Contains "EXT" || case.name.Contains "NV") |> not)
+                    |> Option.orElse (cases |> Array.tryHead)
+                )
+            let res =
+                { groupName = group
+                  cases = cases }
+            return! Some res
+        }
+    )
+
+let paramsTypeRegex = new Regex("<param(.*?(group=\"(?<group>.*?)\").*?|.*?)>(?<f>.*?)(<ptype>(?<t>.*?)<\/ptype>(?<b>.*?))?(<name>.+?<\/name>)<\/param>")
+let protoTypeRegex = new Regex("<proto(.*?(group=\"(?<group>.*?)\").*?|.*?)>(?<f>.*?)(<ptype>(?<t>.*?)<\/ptype>(?<b>.*?))?(<name>.+?<\/name>)<\/proto>")
+let getGroupValue (group: string) (matching: Match) =
+    matching.Groups.[group].Value
+
+let getRawElementDataWithoutLineBreaks (v: Xml.Linq.XElement) =
+    v.ToString().Replace("\n", "").Replace("\t", "").Replace("\r", "")
+
+let extractTypeFromPtype (param: OpenGL_Specification.Param) =
+    let str = getRawElementDataWithoutLineBreaks param.XElement
+    let res = paramsTypeRegex.Match str
+    let groupName =
+        let str = getGroupValue "group" res
+        if str |> String.IsNullOrWhiteSpace then None
+        else Some str
```

Comment:
why not instead 
let groupName = "group" res ||> getGroupValue |> String.IsNullOrEmpty |> Option.defaultValue ""

Reply:
Sometimes it has a group and is related to an enum then but not always therefore its an option